logstash-core/lib/logstash/plugin.rb

```diff
@@ -122,12 +123,22 @@ def metric
                          LogStash::Instrument::NamespacedNullMetric.new(@metric, :null)
                        end
   end
+
   # return the configured name of this plugin
   # @return [String] The name of the plugin defined by `config_name`
   def config_name
     self.class.config_name
   end
 
+  # commit event to dlq
+  def dlq_commit(event, reason)
```

Comment:
I'm not convinced that dlq_commit needs to be available for all plugin types (input, filter, codec, output). I feel like they should only happen for things after the queue?

For example, can you tell me a story about when an input would want to DLQ (At the time of input, we often don't have an event to actually dlq; same goes for codec when we only have bytes, not an event)